### PR TITLE
Preparation to grpc protocol integration

### PR DESCRIPTION
**Pull request by swilgosz**
### Overview

This Branch aggregates all changes to make the gRPC protocol integration possible.

### Todo

- [x] Refactor code to make easier introducing gRPC adapter.
- [x] Fix build
- [x] Add configuration for running the setup with docker.
- [x] Improve configuration mechanism to allow configuring full ES host.
- [x] Move configuration to dry-configurable

**Resolves:**
- [x] #24

### CODE REVIEW

**Comment by KonradBadzioch** [comment]
Why is the branch name `unstable`?

File: lib/event_store_client/configuration.rb [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
+  # Supported adapters: %i[api in_memory]
+  #
+  setting :adapter, :api
 
-    def configure
-      yield(self) if block_given?
-    end
+  setting :error_handler
+  setting :eventstore_url, 'http://localhost:2113' do |value|
+    value.is_a?(String) ? URI(value) : value
```
Comment (KonradBadzioch) [inline lib/event_store_client/configuration.rb:14]: **Question**: Why do you need it here?

**Review by KonradBadzioch — APPROVED** [review]
Look good 🙂  Just one question, I had two, but I checked a commit message and there was explained why you did it. I really love this way of explaining.

File: lib/event_store_client/configuration.rb (hunk 1 above)
Comment (swilgosz) [reply lib/event_store_client/configuration.rb:14]: I want to ensure it's a valid URL. Working with strings makes no sense, as the app will probably crash for invalid data. URI ensures it's at least parsable URL wich reduces the risk of errors. However, there is a bug here. I should have the condition reversed: `valude.is_a?(URI) ? value : URI(value)` - this way config will raise an error for invalid data.

**Comment by swilgosz** [comment]
The branch is named: unstable, as it'd not been tested in real-life applications just yet.

**Pull request merged**